### PR TITLE
Expose delay link manager

### PR DESCRIPTION
**Pull request by astrofrog**
This can be useful for other tools such as jdaviz

### CODE REVIEW

File: glue/core/data_collection.py [hunk 1]
```diff
@@ -126,15 +126,22 @@ def _sync_link_manager(self):
...
 
     @contextmanager
-    def _no_sync_link_manager(self):
+    def _ignore_link_manager_update(self):
         self._disable_sync_link_manager = True
         yield
         self._disable_sync_link_manager = False
 
+    @contextmanager
+    def delay_link_manager_update(self):
```
Comment (pllim) [inline glue/core/data_collection.py:139]: There is no docstring. Will this show up in API doc?

**Review by pllim — COMMENTED** [review]
Maybe another PR, but I think Glue could use a "performance tips and tricks" section like `astropy`. There are some things we could have done to speed things up but they are not intuitive.

File: glue/core/data_collection.py (hunk 1 above)
Comment (astrofrog) [reply glue/core/data_collection.py:139]: Added

**Review by pllim — APPROVED** [review]
LGTM!

**Review by pllim — COMMENTED** [review]
p.s. Bonus would be an example code snippet showing the usage... but it does not block the merge of this PR.

**Pull request merged**